design/one-pager-stack-relationship-labels.md

```diff
@@ -0,0 +1,49 @@
+# Stack Parent/Child Relationship Label Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#934](https://github.com/crossplaneio/crossplane/issues/934)
+
+## Problem
+
+There is currently no easy way to get a list of all resources that are owned by a parent resource. We can leverage CRD labels to make this easier. As long as we have a standard format for doing so.
+
+Per the proposal, it seems that the clearest example of what I am referring to is shown on this page https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels within the `app.kubernetes.io/part-of` label.
+
+As long as we define this label on all children CRD's, we should then be able to query all children of a given parent name using those labels.
+
+## Design
+
+### Example `stack.yaml`
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: Stack
+metadata:
+ name: example-app-name
+...
+```
+
+### Example `crd.yaml`
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+    core.crossplane.io/part-of: "example-app-name"
```

Comment:
If I'm reading the document correctly, it'd make sense to include all of the "recommended labels", so that a query could select on multiple of them. What are your thoughts?

Reply:
I am ok with doing this as long as it doesn't feel too "heavy-handed".

Reply:
I think it seems fair to assume the following if we include more of the recommended labels:

```yaml
app.kubernetes.io/name: mysql
```
`name` would be the current crd

```yaml
app.kubernetes.io/instance: wordpress-abcxzy
```
`instance` would be the current instance of the current part-of stack

```yaml
app.kubernetes.io/version: "5.7.21"
```
`version` would be the current crd's semantic version

```yaml
app.kubernetes.io/component: database
```
`component` would be the current crd's category

```yaml
app.kubernetes.io/part-of: wordpress
```
`part-of` would be the current crd's parent stack

Reply:
Do we also need to include a `namespace` so a reference can cross namespace boundaries?

Reply:
per zoom call, this is the more preferred pattern to go with for this specific use case. We will reserve the _recommended labels_ for when we want to explore that particular use case
```yaml
core.crossplane.io/parent-kind: 'kind'
core.crossplane.io/parent-name: 'name'
core.crossplane.io/parent-namespace: 'namespace'
```

Using this new pattern for a more real example:
```yaml
core.crossplane.io/parent-kind: 'wordpress.samples.crossplane.io'
core.crossplane.io/parent-name: 'wordpress-blue'
core.crossplane.io/parent-namespace: 'xyz'
```